builders/interact/verify-contracts/etherscan-plugins.md

```diff
@@ -0,0 +1,125 @@
+
+![Add an API key](/images/builders/interact/verify-contracts/etherscan-plugins/plugins-2.png)
+
+You will then be prompted to enter in an **AppName** for your API key and once you enter a name and click **Continue** it will appear in your list of API keys. 
+
+## Using the Hardhat Etherscan Plugin {: #using-the-hardhat-etherscan-plugin }
+
+The example in this section of the guide will be based off of the `Box.sol` contract that was created in the [Using Hardhat to Deploy To Moonbeam](/builders/interact/hardhat/){target=blank} guide. 
+
+To get started with the Hardhat Etherscan plugin, you will need to first install the plugin library:
+
+```
+npm install --save-dev @nomiclabs/hardhat-etherscan
+```
+
+You can add your Moonscan API key to the `secrets.json` file alongside your private key.
+
+From within your Hardhat project, open your `hardhat.config.js` file. You'll need to import the `hardhat-etherscan` plugin, your Moonscan API key, and add the config for Etherscan:
+
+```js
+require("@nomiclabs/hardhat-etherscan");
+
+const { privateKey, moonscanApiKey } = require('./secrets.json');
+
+module.exports = {
+  networks: {
+    moonbase: { ... }
+  },
+  etherscan: {
+    apiKey: moonscanApiKey
+  }
+};
+```
+
+To verify the contract, you will run the `verify` command and pass in the address of the deployed contract and the network where it's deployed:
+
+```
+npx hardhat verify --network moonbase <CONTRACT-ADDRESS>
+```
+
+In your terminal you should see the source code for your contract was successfully submitted for verification. If the verification was successful, you should see **Successfully verified contract** and there will be a link to the contract code on [Moonscan for Moonbase Alpha](https://moonbase.moonscan.io/){target=blank}.
+
+![Successful verification using hardhat-etherscan plugin](/images/builders/interact/verify-contracts/etherscan-plugins/plugins-3.png)
+
+If you're verifying a contract that has constructor arguments, you'll need to run the above command and add the constructor arguments used to deploy the contract at the end of the command. For example:
+
+```
+npx hardhat verify --network moonbase <CONTRACT-ADDRESS> "<constructor argument>"
+```
+
+Please refer to the [Hardhat Etherscan documentation](https://hardhat.org/plugins/nomiclabs-hardhat-etherscan.html){target=blank} if you're trying to verify a contract with [complex arguments](https://hardhat.org/plugins/nomiclabs-hardhat-etherscan.html#complex-arguments){target=blank} or [libraries with undetectable addresses](https://hardhat.org/plugins/nomiclabs-hardhat-etherscan.html#libraries-with-undetectable-addresses){target=blank}. Or if you want to use [multiple API keys](https://hardhat.org/plugins/nomiclabs-hardhat-etherscan.html#multiple-api-keys-and-alternative-block-explorers){target=blank} or use the [`verify` command programmatically](https://hardhat.org/plugins/nomiclabs-hardhat-etherscan.html#using-programmatically){target=blank}. 
```

Comment:
I think there are lots of links here we need to do like a bullet list. I would also add this one which has been super helpful lately. This is when the constructor arguments can't be fetched correctly automatically: https://info.etherscan.com/determine-correct-constructor-argument-during-source-code-verification-on-etherscan/